From: Herman van Hazendonk <github.com@herrie.org>
Date: Wed, 26 Aug 2026 10:16:00 +0200
Subject: [PATCH] com.webos.service.systemservice - Allow cardshell to query settings

surfacemanager-cardshell needs systemsettings.query (e.g. wallpaper)
but the group alone isn't enough at oem trust: add dev too. Also add
inbound:["*"] - without it, callers below oem trust get rejected
before the group/client-permission grants are even checked.

Upstream-Status: Pending

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
diff --git a/files/sysbus/com.webos.service.systemservice.groups.json b/files/sysbus/com.webos.service.systemservice.groups.json
--- a/files/sysbus/com.webos.service.systemservice.groups.json
+++ b/files/sysbus/com.webos.service.systemservice.groups.json
@@ -1,7 +1,7 @@
 {
     "allowedNames" : ["com.webos.service.systemservice"],
     "systemsettings.management" : ["oem"],
-    "systemsettings.query" : ["oem"],
+    "systemsettings.query" : ["oem", "dev"],
     "software.query" : ["dev"],
     "time.management" : ["oem"],
     "time.query" : ["dev"],
diff --git a/files/sysbus/com.webos.service.systemservice.perm.json b/files/sysbus/com.webos.service.systemservice.perm.json
--- a/files/sysbus/com.webos.service.systemservice.perm.json
+++ b/files/sysbus/com.webos.service.systemservice.perm.json
@@ -2,5 +2,8 @@
 	"com.webos.service.systemservice":[
 		 "settings.query", "application.launcher", "systemsettings.management",
 		 "networkconnection.query", "telephony.management", "telephony.query"
+	],
+	"com.webos.surfacemanager-cardshell":[
+		 "systemsettings.query"
 	]
 }
diff --git a/files/sysbus/com.webos.service.systemservice.service.role.json.in b/files/sysbus/com.webos.service.systemservice.service.role.json.in
--- a/files/sysbus/com.webos.service.systemservice.service.role.json.in
+++ b/files/sysbus/com.webos.service.systemservice.service.role.json.in
@@ -6,6 +6,7 @@
 	"permissions": [
 		{
 			"service":"com.webos.service.systemservice",
+			"inbound":["*"],
 			"outbound":[
 				"com.webos.service.connectionmanager",
 				"com.webos.service.settingsservice",
